src/entities/lead.js

```diff
@@ -93,3 +99,115 @@ export const getFiltersForRequest = (filters) => {
     });
     return requestFilters;
 };
+
+export const getCombinedLeadFilters = (filters, widgets = [], geoOptions = {}) => {
+    const leadsFilter = {};
+    const entriesFilter = [];
```

Comment:
It would be best if we didn't have to mutate thiese.

Algo:
1. separate out lead and and entries filter
2. useListToMap to get transformed properties
3. merge those intermediate properties.